code401challenges/src/main/java/code401challenges/ArrayShift.java

```diff
@@ -0,0 +1,24 @@
+package code401challenges;
+
+public class ArrayShift {
+    public static int[] insertShiftArray(int[] array, int number) {
+        int[] newArray = new int[array.length + 1];
+        int midpoint = 0;
+        if ((array.length % 2) == 0) {
+            midpoint = array.length/2;
+        } else {
+            midpoint = (array.length + 1)/2;
+        }
```

Comment:
Fun fact: You could always set `midpoint = (array.length + 1) / 2;` and that's correct for both even and odd length arrays!